src/null-safety/index.md

```diff
@@ -1,39 +1,247 @@
+{{ site.alert.end }}
+
+[initializer list]: /guides/language/language-tour#initializer-list
+{% endcomment %}
+
+
+## Understanding list, set, and map types
+
+Lists, sets, and maps are commonly used collection types in Dart programs,
+so you need to know how they interact with null safety.
+Here are some examples of how Dart code uses these collection types:
+
+* Flutter layout widgets such as [`Column`][]
+  often have a `children` property that’s a
+  `List` of `Widget` objects.
+* The [Veggie Seasons][] sample uses a `Set` of `VeggieCategory` to
+  store a user’s food preferences.
+* The [GitHub Dataviz][] sample has a `fromJson()`method that
+  creates an object from JSON data that’s supplied in a
+  `Map<String, dynamic>`.
+
+[`Column`]: https://api.flutter.dev/flutter/widgets/Column-class.html
+[Veggie Seasons]: https://github.com/flutter/samples/tree/master/veggieseasons
+[GitHub Dataviz]: https://github.com/flutter/samples/tree/master/web/github_dataviz
+
+
+### List and set types {#list-and-set-types}
+
+When you’re declaring the type of a list or set, think about what can be null. The following table shows the possibilities for a list of strings
+if you opt into null safety.
+
+{% assign yes = '<b>Yes</b>' %}
+
+|------------------+----------------------------------+--------------------------+------------|
+|Type              | Can an item (string)<br>be null? | Can the list<br>be null? | Description|
+|------------------|----------------------------------|--------------------------|------------|
+| `List<String>`   | No      | No      | A non-null list that contains<br> non-null strings       |
+| `List<String>?`  | No      | {{yes}} | A non-null list that contains<br> strings that **might be null** |
+| `List<String?>`  | {{yes}} | No      | A list that **might be null** and that<br> contains non-null strings |
+| `List<String?>?` | {{yes}} | {{yes}} | A list that **might be null** and that<br> contains strings that **might be null**<br> (same as `List<String>` if you <br>don't opt into null safety) |
+{:.table .table-striped}
+
+When literals initialize a list or set,
+then instead of seeing a type like in the table above,
+you typically see a type annotation on the literal.
+For example, here’s what a `List<String?>` (called `nameList`) and
+a `Set<String?>` (called `nameSet`) might look like:
+
+```dart
+var nameList = <String?>['Andrew', 'Anjan', 'Anya'];
+var nameSet = <String?>{'Andrew', 'Anjan', 'Anya'};
+```
+
+### Map types {#map-types}
+
+Map types behave mostly like you’d expect, with one exception: **the returned value can be null**. Null is the value for a key that isn't present in the map.
+
+As an example, look at the following code. What do you think the value and type of `uhOh` are?
+
+```dart
+var myMap = <String, int>{'one': 1};
+var uhOh = myMap['two'];
+```
+
+The answer is that `uhOh` is null and has type `int?`. 
+
+Like lists and sets, maps can have a variety of type permutations.
```

Comment:
type permutations -> types

Not sure what is being *permuted* here. And "variety of type variants" sounds odd :)